modules/farm_rothamsted_quick/farm_rothamsted_quick.post_update.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+/**
+ * @file
+ * Update hooks for farm_rothamsted.module.
+ */
+
+use Drupal\views\Entity\View;
+use Symfony\Component\Yaml\Yaml;
+
+/**
+ * Create rothamsted_quick_logs view.
+ */
+function farm_rothamsted_quick_post_update_create_rothamsted_quick_logs_view_teo(&$sandbox = NULL) {
+
+  // Only create the view if views is enabled.
+  $view_id = 'rothamsted_quick_logs';
+  if (\Drupal::moduleHandler()->moduleExists('views') && !View::load($view_id)) {
+    $config_path = \Drupal::service('extension.list.module')->getPath('farm_rothamsted_quick') . "/config/install/views.view.$view_id.yml";
+    $data = Yaml::parseFile($config_path);
+    \Drupal::configFactory()->getEditable("views.view.$view_id")->setData($data)->save(TRUE);
+  }
+}
```

Comment:
@mstenta I would like your eyes on this. We need an update hook because we are providing a new (required) view for the `farm_rothamsted_quick` module which is already enabled on deployed sites. Config that is added to `config/install` won't be created for these existing sites.

I found this approach here: https://drupal.stackexchange.com/questions/194679/putting-new-configuration-files-in-module-hook-update-n-upgrade

Tested and it is working but want to run it by you as well. I don't believe we've had to do this type of update for any farmOS 2.x sites up to now?

Reply:
`farm_rothamsted_quick_post_update_create_rothamsted_quick_logs_view_teo`

Oh, this function name is weird because I had to run the update a couple times during testing... let me fix this.

Reply:
That approach should work fine - especially for a custom module like this where we have full control over all deployments. It would probably work fine for core too - we just need to be a little more careful. The main risk (that I know of) is described in the `hook_update_N()` docs:

> ### Notes about the function body
> Writing [hook_update_N](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Extension%21module.api.php/function/hook_update_N/9.2.x)() functions is tricky. There are several reasons why this is the case:
> ...

(see https://api.drupal.org/api/drupal/core!lib!Drupal!Core!Extension!module.api.php/function/hook_update_N/9.2.x#sec_notes)

In particular:

> **You do not know when updates will be run**: someone could be keeping up with every update and run them when the database and code are in the same state as when you wrote your update function, or they could have waited until a few more updates have come out, and run several at the same time.

And therefore (in the context of config specifically):

> Never assume that the configuration schema is the same when the update will run as it is when you wrote the update function.

That doc page is worth reading and digesting in its entirety. It describes situations that may not come to mind when writing update hooks, but which need to be accounted for (unless you are the only one deploying the updates yourself, and you have full knowledge of the state of the site, as we do in this case).

I'm surprised that the `hook_post_update_NAME()` docs don't explicitly reference and link to those considerations.Many (not all) of the considerations apply to both.

Reply:
Good thoughts!